server/src/main/java/org/opensearch/ingest/IngestPipelineValidator.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.ingest;
+
+import java.util.List;
+
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Setting;
+
+/**
+ * This class contains methods to validate the ingest pipeline.
+ */
+public class IngestPipelineValidator {
+
+    /**
+     * Defines the limit for the number of processors which can run on a given document during ingestion.
+     *
+     */
+    public static final Setting<Integer> MAX_NUMBER_OF_INGEST_PROCESSORS = Setting.intSetting(
+        "cluster.ingest.max_number_processors",
+        Integer.MAX_VALUE,
+        1,
+        Integer.MAX_VALUE,
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
+
+    /**
+     * Validates that the number of compound processors in the pipeline does not exceed the configured limit.
+     * @param pipeline
+     * @param clusterService
+     */
+    public static void validateIngestPipeline(Pipeline pipeline, ClusterService clusterService) {
+
+        List<Processor> processors = pipeline.getCompoundProcessor().getProcessors();
+        int maxNumberOfIngestProcessorsAllowed = clusterService.getClusterSettings().get(MAX_NUMBER_OF_INGEST_PROCESSORS);
```

Comment:
We can submit the settingUpdateConsumer for the setting, and avoid reading setting from cluster state every time. With that we wont need the cluster service in validate method as well and simplify some thing on Simulate API side.

Reply:
Updated.